data/option.lean

```diff
@@ -100,4 +100,26 @@ def to_list : option α → list α
 @[simp] theorem mem_to_list {a : α} {o : option α} : a ∈ to_list o ↔ a ∈ o :=
 by cases o; simp [to_list, eq_comm]
 
+def lift_or_get (f : α → α → α) : option α → option α → option α
```

Comment:
How about calling this `merge`? `lift_or_get` is a bit of a mouthful.

Reply:
The name `get_or_else` in turn is borrowed from Rust terminology. What other operations are you thinking about? I think all the obvious ones are already defined and named; in particular `<|>` is the only other reasonable merge-like operation (which is equivalent to` `lift_or_get (\lam x y, x)`) I can think of, so I don't think future name clashes are likely. What about `merge_with`?

Reply:
Well, there's the “obvious” function with the type `(α → β → γ) → option α → option β → option γ`, which could be considered a `merge` or `merge_with`. However, I'd call that function something like `option.lift₂`. In that sense, I think `option.lift_or_get` (or `option.get_or_lift`, `option.lift₂_or_get`, or some other variation) is a more descriptive name than `merge`/`merge_with` (or [`combineMaybe`](https://hackage.haskell.org/package/toysolver-0.5.0/docs/src/ToySolver-Internal-Util.html#combineMaybe) or [`maybe_join`](https://hackage.haskell.org/package/hmt-0.16/docs/src/Music-Theory-Maybe.html#maybe_join) found in various Haskell projects), since you have nods to both `get_or_else` and `lift`. But I don't care so much as to make a fuss about the name. In the end, you're welcome to choose one you like.